plugins/gradle-plugin/src/main/kotlin/com/android/designcompose/gradle/PluginExtension.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.designcompose.gradle
+
+import org.apache.tools.ant.taskdefs.condition.Os
+import org.gradle.api.Project
+import org.gradle.api.file.Directory
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.Provider
+
+private const val tokenGradlePropertyName = "figmaAccessToken"
+private const val tokenFileGradlePropertyName = "figmaAccessTokenFilePath"
+
+/**
+ * Plugin extension for the DesignCompose plugin
+ *
+ * Holds persistent data for the plugin including the path to the user's Figma Token File and the
+ * user's token itself.
+ */
+abstract class PluginExtension() {
+    abstract val figmaTokenFile: RegularFileProperty
+    abstract val figmaToken: Property<String>
+}
+
+/**
+ * Initialize the plugin extension to set conventions (default values) for the properties
+ *
+ * @param extension: The instance of the extension that was created with project.extensions.create
+ */
+fun Project.initializeExtension(extension: PluginExtension) {
+
+    // Determine the directory where the token file should live, based on OS
+    val tokenFileDir: Provider<Directory> =
+        if (Os.isFamily(Os.FAMILY_WINDOWS)) {
```

Comment:
consider using when() , see [this](https://kotlinlang.org/docs/control-flow.html#when-expression)

Reply:
🤦